site/src/pages/WorkspacePage/WorkspaceNotifications.tsx

```diff
@@ -0,0 +1,343 @@
+      ) : undefined,
+    });
+  }
+
+  // Dormant
+  const areActionsEnabled = useIsWorkspaceActionsEnabled();
+  if (areActionsEnabled && workspace.dormant_at) {
+    const formatDate = (dateStr: string, timestamp: boolean): string => {
+      const date = new Date(dateStr);
+      return date.toLocaleDateString(undefined, {
+        month: "long",
+        day: "numeric",
+        year: "numeric",
+        ...(timestamp ? { hour: "numeric", minute: "numeric" } : {}),
+      });
+    };
+    const actions = (
+      <NotificationActionButton onClick={onActivateWorkspace}>
+        Activate
+      </NotificationActionButton>
+    );
+    notifications.push({
+      actions,
+      title: "Workspace is dormant",
+      severity: "warning",
+      detail: workspace.deleting_at ? (
+        <>
+          This workspace has not been used for{" "}
+          {formatDistanceToNow(Date.parse(workspace.last_used_at))} and was
+          marked dormant on {formatDate(workspace.dormant_at, false)}. It is
+          scheduled to be deleted on {formatDate(workspace.deleting_at, true)}.
+          To keep it you must activate the workspace.
+        </>
+      ) : (
+        <>
+          This workspace has not been used for{" "}
+          {formatDistanceToNow(Date.parse(workspace.last_used_at))} and was
+          marked dormant on {formatDate(workspace.dormant_at, false)}. It is not
+          scheduled for auto-deletion but will become a candidate if
+          auto-deletion is enabled on this template. To keep it you must
+          activate the workspace.
+        </>
+      ),
+    });
+  }
+
+  // Pending in Queue
+  const [showAlertPendingInQueue, setShowAlertPendingInQueue] = useState(false);
+  // 2023-11-15 - MES - This effect will be called every single render because
+  // "now" will always change and invalidate the dependency array. Need to
+  // figure out if this effect really should run every render (possibly meaning
+  // no dependency array at all), or how to get the array stabilized (ideal)
+  const now = dayjs();
+  useEffect(() => {
+    if (
+      workspace.latest_build.status !== "pending" ||
+      workspace.latest_build.job.queue_size === 0
+    ) {
+      if (!showAlertPendingInQueue) {
+        return;
+      }
+
+      const hideTimer = setTimeout(() => {
+        setShowAlertPendingInQueue(false);
+      }, 250);
+      return () => {
+        clearTimeout(hideTimer);
+      };
+    }
+
+    const t = Math.max(
+      0,
+      5000 - dayjs().diff(dayjs(workspace.latest_build.created_at)),
+    );
+    const showTimer = setTimeout(() => {
+      setShowAlertPendingInQueue(true);
+    }, t);
+
+    return () => {
+      clearTimeout(showTimer);
+    };
+  }, [workspace, now, showAlertPendingInQueue]);
+
+  if (showAlertPendingInQueue) {
+    notifications.push({
+      title: "Workspace build is pending",
+      severity: "info",
+      detail: (
+        <>
+          <div css={{ marginBottom: 12 }}>
```

Comment:
Could this be turned into a `<p>`?

Reply:
It already wraps the content into a p element. Going just to remove the div and replace the second p by an <span>